client/my-sites/site-settings/redirect-to/redirect-to.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpack = ( WrappedComponent, redirectUrl ) => {
+	class RedirectNonJetpack extends Component {
+		static propTypes = {
+			redirectUrl: PropTypes.string,
+			// Connected props
+			siteIsAtomic: PropTypes.bool,
+			siteIsJetpack: PropTypes.bool,
+			siteSlug: PropTypes.string
+		}
+
+		componentDidMount() {
+			this.verifySiteIsJetpackOrAtomic();
+		}
+
+		componentDidUpdate() {
+			this.verifySiteIsJetpackOrAtomic();
+		}
+
+		verifySiteIsJetpackOrAtomic() {
+			if ( this.props.siteIsJetpack === false || this.props.siteIsAtomic ) {
+				this.redirectTo();
+			}
+		}
+
+		redirectTo = () => {
+			const { siteSlug } = this.props;
+
+			if ( siteSlug ) {
+				const url = redirectUrl ? redirectUrl : '/settings/general/';
+
+				page( url + siteSlug );
```

Comment:
We can't be certain that any custom `redirectUrl ` route will end with the site slug. So instead we should probably put the `url + siteSlug` part only in case we use the default `redirectUrl`, and not when it's specifically passed to the HoC as a prop.